pkg/ccl/backupccl/backup_planning.go

```diff
@@ -1394,6 +1401,93 @@ func backupPlanHook(
 	return fn, utilccl.BulkJobExecutionResultHeader, nil, false, nil
 }
 
+func getScheduledBackupExecutionArgsFromSchedule(
+	ctx context.Context,
+	env scheduledjobs.JobSchedulerEnv,
+	txn *kv.Txn,
+	ie *sql.InternalExecutor,
+	scheduleID int64,
+) (*jobs.ScheduledJob, *ScheduledBackupExecutionArgs, error) {
+	// Load the schedule that has spawned this job.
+	sj, err := jobs.LoadScheduledJob(ctx, env, scheduleID, ie, txn)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to load scheduled job %d", scheduleID)
+	}
+
+	args := &ScheduledBackupExecutionArgs{}
+	if err := pbtypes.UnmarshalAny(sj.ExecutionArgs().Args, args); err != nil {
+		return nil, nil, errors.Wrap(err, "un-marshaling args")
+	}
+
+	return sj, args, nil
+}
+
+func setupProtectedTimestampChaining(
+	ctx context.Context,
+	p sql.PlanHookState,
+	backupDetails *jobspb.BackupDetails,
+	backupStmt *annotatedBackupStatement,
+) error {
+	env := scheduledjobs.ProdJobSchedulerEnv
+	if knobs, ok := p.ExecCfg().DistSQLSrv.TestingKnobs.JobsTestingKnobs.(*jobs.TestingKnobs); ok {
+		if knobs.JobSchedulerEnv != nil {
+			env = knobs.JobSchedulerEnv
+		}
+	}
+	// If this is not a scheduled backup, we do not chain pts records.
+	if backupStmt.CreatedByInfo == nil || backupStmt.CreatedByInfo.Name != jobs.CreatedByScheduledJobs {
+		return nil
+	}
+
+	sj, args, err := getScheduledBackupExecutionArgsFromSchedule(ctx, env,
```

Comment:
Does this need to be done during planning time? Would it be reasonable to just persist the created by info ID in the job details?

Reply:
I assume you mean why we need to identify the record that needs to be updated/released in planning rather than job execution?

The way I thought about this was that doing this in the planners' txn gives us the guarantee that it'll be done exactly once when the job run as part of the schedule is being spawned. If we put this in the job we have to worry about resumptions since we do not want to read a new record to release every time we resume. We can ofcourse enforce idempotency like we do elsewhere but hypothetically,

- Inc A is planned
- Inc A begins execution and picks the prev incs record to update
- Full A is planned
- Full A begins execution
- Full A completes execution
- Inc A is resumed thereby picking the pts record written by Full A on the inc schedule
- Inc A completes incorrectly pulling forward the pts record.

If I'm misunderstanding what you're saying let me know.

Reply:
Yep, that makes sense. Thanks for clarifying.